cmake/PcapPlusPlusConfig.cmake.in

```diff
@@ -0,0 +1,29 @@
+# ~~~
+# - Config file for the PcapPlusPlus package
+# It defines the following variables
+#  PcapPlusPlus_INCLUDE_DIRS - include directories for PcapPlusPlus
+#  PcapPlusPlus_LIBRARIES    - libraries to link against
+# ~~~
+
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+set_and_check(PcapPlusPlus_INCLUDE_DIR
+              ${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_INCLUDEDIR@/pcapplusplus)
+
+if(UNIX)
+  set(THREADS_PREFER_PTHREAD_FLAG ON)
+  find_dependency(Threads REQUIRED)
```

Comment:
Could you explain a bit more please? I don't understand which possibilities are limited which this REQUIRED. Thanks :)

Reply:
Read the cmake doc understood, thanks!

Reply:
`find_dependency` already implies that the the given package is required. If the dependency can't be resolved, this particular `PcapPlusPlusConfig.cmake` file is discarded. The user project may eventually receive a `PcapPlusPlus_FOUND` with a false value.

If you do `find_dependency(... REQUIRED)`, it will terminate CMake immediately, even for user projects which don't strictly require `find_package(PcapPlusPlus)`.

(Note how your `CMakeLists.txt` avoids `find_package(<Pkg> REQUIRED)` even for clearly required packages (canonical usage), just to give a different type of error message for `NOT <Pkg>_FOUND`.)